src/include/duckdb/parser/column_definition.hpp

```diff
@@ -96,6 +104,8 @@ class ColumnDefinition {
 	TableColumnType category = TableColumnType::STANDARD;
 	//! Used by Generated Columns
 	unique_ptr<ParsedExpression> generated_expression;
+	//! Flag to define a serial column
+	bool is_serial = false;
```

Comment:
this is heavily inspired by the Postgres implementation, see https://github.com/postgres/postgres/blob/b0a55e43299c4ea2a9a8c757f9c26352407d0ccc/src/backend/parser/parse_utilcmd.c#L539

serial is definitely not a type, but it is not really a constraint either is it? it's just an integer with a not null constraint, so something as simple as a flag seemed a reasonable way to pass this information around